scc/light_client/provider/server.go

```diff
@@ -0,0 +1,167 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/0xsoniclabs/sonic/ethapi"
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+	idx "github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// Server implements the Provider interface and provides methods
+// making RPC calls through an RPC client.
+type Server struct {
+	// client is the RPC client used for making RPC calls.
+	client RpcClient
+}
+
+// NewServerFromClient creates a new Server with the given
+// RPC client. The resulting Server takes ownership of the client and
+// will close it when the Server is closed.
+// The resulting Server must be closed after use.
+//
+// Parameters:
+// - client: The RPC client to use for RPC calls.
+//
+// Returns:
+// - *Server: A new instance of Server.
+// - error: An error if the client is nil.
+func NewServerFromClient(client RpcClient) (*Server, error) {
```

Comment:
I guess this factory is not that useful in the long run, since the anticipated use case is satisfied by `NewServerFromURL`. Do you think it can/should be dropped? Maybe as a follow-up.

This would also remove complexity from integration tests.

Reply:
I thought about changing `NewServerFromURL` into the default `NewServer(url)` but this can also be a follow-up. 
As for dropping the `FromClient` alternative, I thought it would be nice as an alternative for other `RpcClient` implementations.